app/main.py

```diff
@@ -1,2 +1,140 @@
+from __future__ import annotations
+from typing import Hashable, Any, Iterator, Iterable
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Pair:
+    key: Hashable
+    value: Any
+    key_hash: int
+
+
+BucketType = list[Pair]
+
+
 class Dictionary:
-    pass
+
+    DEFAULT_CAPACITY: int = 8
+
+    def __init__(
+            self,
+            capacity: int = DEFAULT_CAPACITY,
+            load_factor_limit: float = 0.75
+    ) -> None:
+        self._buckets: list[BucketType] = [[] for _ in range(capacity)]
+
+        self._load_factor_limit = load_factor_limit
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self._load_factor() >= self._load_factor_limit:
+            self._resize_and_rehash()
+
+        index = self._index(key)
+        bucket = self._buckets[index]
+        try:
+            index_in_bucket, _ = self._find_pair(bucket, key)
+            bucket[index_in_bucket] = Pair(key, value, index)
+        except KeyError:
+            bucket.append(Pair(key, value, index))
+
+    def __getitem__(self, key: Hashable) -> Any:
+        bucket = self._buckets[self._index(key)]
+        _, pair = self._find_pair(bucket, key)
+        return pair.value
+
+    def __delitem__(self, key: Hashable) -> None:
+        bucket = self._buckets[self._index(key)]
+        index, _ = self._find_pair(bucket, key)
+        del bucket[index]
+
+    def __contains__(self, key: Hashable) -> bool:
+        try:
+            self[key]
+        except KeyError:
+            return False
+        else:
+            return True
+
+    def __len__(self) -> int:
+        return len(self.items())
+
+    def __iter__(self) -> Iterator:
+        yield from self.keys()
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def items(self) -> list[tuple[Hashable, Any]]:
+        return [
+            (pair.key, pair.value) for bucket in self._buckets
+            for pair in bucket
+        ]
+
+    def values(self) -> list[Any]:
+        return [pair[1] for pair in self.items()]
+
+    def keys(self) -> list[Hashable]:
+        return [pair[0] for pair in self.items()]
```

Comment:
The `keys` method incorrectly accesses tuple elements using indexing. Instead of `pair[0]`, use `pair.key` to access the key directly from the `Pair` object.